inc/namespace.php

```diff
@@ -590,3 +591,55 @@ function get_report_only_content_security_policies() : array {
 	 */
 	return apply_filters( 'altis.security.browser.report_only_content_security_policies', $policies );
 }
+
+/**
+ * Restrict CORS origin
+ * 
+ * @return mixed
+ */
+function restrict_cors_origin( $result ) : mixed {
+	$rest_allow_origin = apply_filters( 'altis.security.browser.rest_allow_origin', $value );
+	// Remove CORS header
+	if ( ! $rest_allow_origin ) {
+		// Response 403 if origin is explicitly not allowed
+		return new WP_Error( 'forbidden_access', 'Access Forbidden', array( 'status' => 403 ) );
+	} elseif ( $rest_allow_origin ) {
+		$allowed_cors_origins = get_allowed_cors_origins();
+		if ( ! empty( get_allowed_cors_origins() ) ) {
+			add_filter( 'http_origin', __NAMESPACE__ . '\\check_if_cors_origin_allowed' );
+		}
+	}
+
+	return $result;
```

Comment:
This code currently looks like:

```php
if ( ! $bool ) {
    // 1...
} elseif ( $bool ) {
    // 2...
}
// 3...
```

The third case here will never actually happen since the bool is either true or false. In addition, it can be adapted to use the return-early pattern instead.

Some other minor tweaks too (you'll need to add `use WP_Http` to the top of the file):

```suggestion
	$rest_allow_origin = apply_filters( 'altis.security.browser.rest_allow_origin', $value );

	if ( ! $rest_allow_origin ) {
		return new WP_Error( 'altis.security.browser.origin_not_allowed', 'Origin is not on allowed list', [ 'status' => WP_Http::FORBIDDEN ] );
	}

	$allowed_cors_origins = get_allowed_cors_origins();
	if ( ! empty( get_allowed_cors_origins() ) ) {
		add_filter( 'http_origin', __NAMESPACE__ . '\\check_if_cors_origin_allowed' );
	}

	return $result;
```

Now that I read it again, I'm also not sure on why there are two filters here (origin vs origins)?

Reply:
@rmccue Regarding `origin` and `origins`, my understanding with the [acceptance criteria](https://github.com/humanmade/altis-browser-security/issues/19) is to implement a filter to enable or disable CORS headers, and another filter to set the allowed origins. Does that make sense?

Should this be combined to a single filter? 
- If `origins` is true, keep default WordPress behaviour
- If `origins` is false, respond 403
- If `origins` is an array, use it as list of origins allowed

Reply:
Ah, that's definitely unclear then. The idea was that you would be able to set a fixed list of origins, and also dynamically control the behaviour with a filter, where that filter receives the value.

For example, you might want to do something like
```php
add_filter( 'altis.security.browser.rest_allow_origin', function ( $allow, $origin ) {
    // Disallow all .local domains
    if ( substr( $origin, '.local' ) >= 0 ) {
        return false;
    }

    return $allow;
}, 10, 2 );
```

It's not possible to implement that functionality statically, so it has to be a dynamic filter.